[metro][expo-updates] updated imports in expo-updates

chrfalch opened the pull request:
# Why

After upgrading to Metro 0.83 we changed how/what we can import from Metro. This causes expo-updates build to fail on CI due to some direct imports from Metro.

# How

This commit fixes this by using the (hopefully) correct imports.

# Test Plan

Make sure CI passes

# Checklist

- [x] I added a `changelog.md` entry and rebuilt the package sources according to [this short guide](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-before-submitting)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

chrfalch committed:
[metro][expo-updates] updated imports in expo-updates

After upgrading to Metro 0.83 we changed how/what we can import from Metro. This causes expo-updates build to fail on CI due to some direct imports from Metro.

This commit fixes this by using the (hopefully) correct imports.

chrfalch committed:
Added changelog entry

chrfalch committed:
Update packages/expo-updates/CHANGELOG.md

Co-authored-by: Expo Bot <34669131+expo-bot@users.noreply.github.com>

byCedric reviewed (changes requested):
`@expo/metro` should be added as a `devDependency` to `expo-updates`.

chrfalch committed:
Added @expo/metro as devDependency to expo-updates

Since we're now importing types and members from @expo/metro we need to add it as a dependency as well.

chrfalch commented:
> `@expo/metro` should be added as a `devDependency` to `expo-updates`.

Thanks - appreciate it, @byCedric <3 Updated now!

Kudo commented on packages/expo-updates/utils/src/createManifestForBuildAsync.ts:
```diff
@@ -3,13 +3,13 @@ import {
 } from '@expo/cli/build/src/export/embed/exportEmbedAsync';
 import { drawableFileTypes } from '@expo/cli/build/src/export/metroAssetLocalPath';
+import Server from '@expo/metro/metro/Server';
+import type { BundleOptions } from '@expo/metro/metro/shared/types.flow';
```
should we try to resolve `expo -> `@expo/metro`? same as the others like `@expo/metro-config`, not sure whether it didn't break pnpm isolated mode from the past. cc @byCedric

Kudo replied:
yes we can merge this and follow up later if that's a problem. still wonder whether it doesn't break from the past haha

byCedric replied:
I think it’s fine, we’ll keep expo metro versioned by Expo SDK

Kudo reviewed (approved):
still delegating Cedric for final review because he has more knowledge for `@expo/metro`
